### PR TITLE
workaround for checkbox not propagating changes to KeepPrivate property

### PR DESCRIPTION
**Pull request by shiftkey**
Proposed fix for #62 by observing the events directly

 - [x] still have to bind the property to the VM somehow.
 - [x] testing testing testing

### CODE REVIEW

File: src/GitHub.VisualStudio/UI/Views/Controls/RepositoryPublishControl.xaml.cs [hunk 1]
```diff
@@ -30,7 +30,13 @@ public RepositoryPublishControl()
...
                 
                 d(this.Bind(ViewModel, vm => vm.Description, v => v.description.Text));
-                d(this.Bind(ViewModel, vm => vm.KeepPrivate, v => v.makePrivate.IsChecked));
+
+                d(Observable.FromEventPattern<RoutedEventArgs>(makePrivate, "Checked")
+                    .Subscribe(_ => ViewModel.KeepPrivate = true));
+
+                d(Observable.FromEventPattern<RoutedEventArgs>(makePrivate, "Unchecked")
+                    .Subscribe(_ => ViewModel.KeepPrivate = false));
+
```
Comment (Haacked) [inline src/GitHub.VisualStudio/UI/Views/Controls/RepositoryPublishControl.xaml.cs]: Let's add a comment on why this abomination is required as opposed to the mores straightforward code that works elsewhere. :smile:

Comment (shiftkey) [inline src/GitHub.VisualStudio/UI/Views/Controls/RepositoryPublishControl.xaml.cs]: :thumbsup:

**Comment by Haacked** [comment]
> does your binding need an UpdateSourceTrigger maybe

@gulbanana We're not using XAML bindings. We're using ReactiveUI bindings. Note that the `Bind` method we used here is the same one we use in the repository create control: https://github.com/github/VisualStudio/blob/master/src/GitHub.VisualStudio/UI/Views/Controls/RepositoryCreationControl.xaml.cs#L50

It works over there, but not here for some reason.

**Comment by shiftkey** [comment]
Ready for review.

**Comment by Haacked** [comment]
:cry:

Thanks for the fix @shiftkey! :sparkles:

**Comment by gulbanana** [comment]
ah, ok. i've definitely experienced weird partial-supportedness with the IsChecked dependencyproperty before but that was using {Binding}, not reactiveui.

**Pull request merged**